include/common/common.h

```diff
@@ -20,6 +20,8 @@
 
 #ifndef GRPPI_COMMON_H
 #define GRPPI_COMMON_H
+//Include standard experimental optional
+#include <experimental/optional>
```

Comment:
I do not think we need to include optional here in any form.

Just include in the places we really use it.